lib/fileio/commands/baseExportCommand.h

```diff
@@ -0,0 +1,109 @@
+//
+// Copyright 2016 Pixar
+// Copyright 2020 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef MAYA_EXPORT_COMMAND_H
+#define MAYA_EXPORT_COMMAND_H
+
+#include "../../base/api.h"
+
+#include <memory>
+
+#include <maya/MPxCommand.h>
+
+#include <mayaUsd/fileio/jobs/writeJob.h>
+
+MAYAUSD_NS_DEF {
+
+class MAYAUSD_CORE_PUBLIC MayaUSDExportCommand : public MPxCommand
+{
+  public:
+    //
+    // Command flags are a mix of Arg Tokens defined in writeJob.h
+    // and some that are defined by this command itself.
+    // All short forms of the Maya flag names are defined here.
+    // All long forms of flags defined by the command are also here.
+    // All long forms of flags defined by the Arg Tokens are queried
+    // for and set when creating the MSyntax object.
+    // Derived classes can use the short forms of the flags when
+    // calling Maya functions like argData.isFlagSet()
+    //
+    // The list of short forms of flags defined as Arg Tokens:
+    static constexpr auto kDefaultMeshSchemeFlag = "dms";
```

Comment:
This is nice of course, but I don't see any corresponding definition for these statics in the .cpp implementation.  Are you sure this compiles on all architectures?  I believe that Sean and I ran into exactly this problem during the development of the UFE attribute base class, and we had to add definitions in the .cpp.  I can show you the (Autodesk internal) code if you'd like.

Reply:
I'll ping you about it

Reply:
@ppt-adsk   I chatted with Sean briefly and I think the difference between this and UFE is that your class there is used in a template where I'm not here.  I can also find plenty of other uses like this in the repo so I think I'm ok here.  

I thought I could put it in there just in case, but then (if I'm understanding it correctly) I'd have to drop the usage of "auto" because I don't think I can just declare it there without assign it anything.

Long story short, I think we're good here, but it's an interesting c++ issue to understand.